schema.json

```diff
@@ -0,0 +1,122 @@
+{
+  "id": "http://schemas.taskcluster.net/generic-worker/v1/payload.json#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Generic worker payload",
+  "description": "This schema defines the structure of the `payload` property referred to in a Task Cluster Task definition.",
+  "type": "object",
+  "required": [
+    "command",
+    "maxRunTime"
+  ],
+  "properties": {
+    "command": {
+      "title": "Command to run.",
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "string"
+      },
+      "description": "Example: `['/bin/bash', '-c', 'build.sh']`."
+    },
+    "encryptedEnv": {
+      "title": "List of encrypted environment variable mappings.",
+      "description": "List of base64 encoded asymmetric encrypted environment variables. See http://docs.taskcluster.net/docker-worker/#encrypted-environment-variables",
+      "type": "array",
+      "items": {
+        "title": "Base64 encoded encrypted environment variable object.",
+        "type": "string"
+      }
+    },
+    "env": {
+      "title": "Environment variable mappings.",
+      "description": "Example: ```\n{\n  \"PATH\": '/borked/path' \n  \"ENV_NAME\": \"VALUE\" \n}\n```",
+      "type": "object"
+    },
+    "maxRunTime": {
+      "type": "number",
+      "title": "Maximum run time in seconds",
+      "description": "Maximum time the task container can run in seconds",
+      "multipleOf": 1.0,
+      "minimum": 1,
+      "maximum": 86400
+    },
+    "graphs": {
+      "type": "array",
+      "title": "Paths (in the container) to a json files which are used to extend the task graph",
+      "description": "Contents of file are used to extend the graph (if this task was part of a graph). See http://docs.taskcluster.net/scheduler/api-docs/#extendTaskGraph",
+      "items": {
+        "title": "Individual path to graph extension point.",
+        "type": "string"
+      }
+    },
+    "artifacts": {
```

Comment:
You don't have to support `artifacts` initially, but it's very very nice to have :)

Either way, if you do I would suggest making it an array of objects instead of a mapping. But as I see it, you can just leave it up to the tasks to upload artifacts for now...